### PR TITLE
Add "Private" Environment Variables

### PR DESCRIPTION
**Pull request by peterallenwebb**
### Description

Support environment variables with a DBT_ENV_PRIVATE_ prefix, which dbt can see, but user code cannot.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-common/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-common/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

File: dbt_common/context.py [hunk 1]
```diff
@@ -1,20 +1,29 @@
...
+from dbt_common.constants import PRIVATE_ENV_PREFIX, SECRET_ENV_PREFIX
 
 
 class InvocationContext:
     def __init__(self, env: Mapping[str, str]):
-        self._env = env
+        self._env = {k: v for k, v in env.items() if not k.startswith(PRIVATE_ENV_PREFIX)}
         self._env_secrets: Optional[List[str]] = None
+        self._env_private = {
+            k[len(PRIVATE_ENV_PREFIX) :]: v
```
Comment (MichelleArk) [inline dbt_common/context.py:12]: what's the motivation for stripping the prefix off of private envs?

Comment (peterallenwebb) [reply dbt_common/context.py:12]: My take is that it makes them less annoying and verbose to look up later, but I could go either way.

Comment (peterallenwebb) [reply dbt_common/context.py:12]: I think even the need to document it tilts me to the side of just keeping the prefix, so I'll just do it that way.

Comment (peterallenwebb) [reply dbt_common/context.py]: Done!

**Review by MichelleArk — APPROVED** [review]

**Pull request merged**